### PR TITLE
[Docs] Warning on metric states

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Based on discussion in slack and in issue https://github.com/PyTorchLightning/pytorch-lightning/issues/4357, it is not clear to users that metrics should be re-initialized per mode (training, validation, testing) such that metric states are not mixed.
This PR adds a warning to the docs.

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.    
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/metrics.rst [hunk 1]
```diff
@@ -105,6 +104,13 @@ This metrics API is independent of PyTorch Lightning. Metrics can directly be us
     # total accuracy over all validation batches
     total_valid_accuracy = valid_accuracy.compute()
 
+.. warning::
+
+    Metrics have internal states that keeps track of the data seen so far. Too
+    not mix metric states across training, validation and testing it is
+    highly recommended to re-initialize the metric per mode as shown in the
+    examples above.
+
```
Comment (SeanNaren) [inline docs/source/metrics.rst:113]: I think there is a spelling mistake here, might be better to reword like below:

Metrics contain internal states that keep track of the data seen so far. Do not mix metric states across training, validation and testing. It is highly recommended to re-initialize the metric per mode as shown in the examples above.

**Review by SeanNaren — CHANGES_REQUESTED** [review]

File: docs/source/metrics.rst (hunk 1 above)
Comment (SkafteNicki) [reply docs/source/metrics.rst:113]: thanks, changed it to your suggestion 👍

File: docs/source/metrics.rst [hunk 2]
```diff
@@ -105,6 +104,13 @@ This metrics API is independent of PyTorch Lightning. Metrics can directly be us
     # total accuracy over all validation batches
     total_valid_accuracy = valid_accuracy.compute()
 
+.. warning::
```
Comment (justusschock) [inline docs/source/metrics.rst:107]: Can we rather make this a note than a warning? I think `warning` would mean that there are some issues on our side.

**Review by justusschock — APPROVED** [review]

File: docs/source/metrics.rst (hunk 2 above)
Comment (SkafteNicki) [reply docs/source/metrics.rst]: You are right, have changed it 👍

**Review by s-rog — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Pull request merged**